zmq.hpp

```diff
@@ -1110,6 +1110,19 @@ const_buffer buffer(std::basic_string_view<T, Traits> data, size_t n_bytes) noex
 }
 #endif
 
+// Buffer for a string literal (null terminated)
+// where the buffer size excludes the terminating character.
+// Equivalent to zmq::buffer(std::string_view("...")).
+template<class Char, size_t N>
+const_buffer str_buffer(const Char (&data)[N]) noexcept
```

Comment:
I am not 100% sure, but couldn't this be constexpr? (Maybe the assert conflicts with that, but then it could be constexpr at least with NDEBUG defined). Maybe the assertion could be avoided entirely if this were a user-defined literal operator instead?